TweetMate/src/main/java/com/ybsystem/tweetmate/usecases/DownloadUseCase.java

```diff
@@ -0,0 +1,107 @@
+package com.ybsystem.tweetmate.usecases;
+
+import android.app.Activity;
+import android.content.ContentResolver;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.net.Uri;
+import android.os.Build;
+import android.os.Environment;
+
+import com.ybsystem.tweetmate.application.TweetMateApp;
+import com.ybsystem.tweetmate.utils.DialogUtils;
+import com.ybsystem.tweetmate.utils.StorageUtils;
+import com.ybsystem.tweetmate.utils.ToastUtils;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.InputStream;
+
+import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers;
+import io.reactivex.rxjava3.core.Observable;
+import io.reactivex.rxjava3.observers.DisposableObserver;
+import io.reactivex.rxjava3.schedulers.Schedulers;
+
+import static com.ybsystem.tweetmate.resources.ResString.*;
+
+public class DownloadUseCase {
+
+    private DownloadUseCase() {
+    }
+
+    public static void downloadImage(String url) {
+        // Async
+        Observable<String> observable = createSaveImageObs(url);
+
+        // Result
+        DisposableObserver<String> disp = new DisposableObserver<String>() {
+            @Override
+            public void onNext(String path) {
+                // Success
+                ToastUtils.showLongToast(STR_SUCCESS_SAVE_IMAGE + "\n" + path);
+            }
+
+            @Override
+            public void onError(Throwable t) {
+                // Failed...
+                ToastUtils.showShortToast(t.getMessage());
+            }
+
+            @Override
+            public void onComplete() {
+            }
+        };
+
+        // Show progress
+        DialogUtils.showProgress(STR_CONNECTING, TweetMateApp.getActivity());
+
+        // Subscribe
+        observable.subscribeOn(Schedulers.newThread())
+                .observeOn(AndroidSchedulers.mainThread())
+                .doFinally(DialogUtils::dismissProgress)
+                .subscribe(disp);
+    }
+
+    private static Observable<String> createSaveImageObs(String url) {
+        // Create
+        return Observable.create(e -> {
+            String path = "";
+            FileOutputStream out = null;
+
+            try {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Change this "try" to a try-with-resources. (sonar.java.source not set. Assuming 7 or greater.) [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://192.168.11.19:9000/coding_rules#rule_key=squid%3AS2093)